components/elm/src/biogeochem/AllocationMod.F90

```diff
@@ -2335,9 +2343,9 @@ subroutine Allocation3_PlantCNPAlloc (bounds            , &
              else if ( carbonphosphorus_only ) then
                  N_lim_factor(p) = 0.0_r8
              end if
-             W_lim_factor(p) = 0.0_r8
+             water_scalar(p) = 0.0_r8
              do j = 1 , nlevdecomp
-                 W_lim_factor(p) = W_lim_factor(p) + w_scalar(c,j) * froot_prof(p,j)
+                 water_scalar(p) = water_scalar(p) + w_scalar(c,j) * froot_prof(p,j) * dzsoi_decomp(j)
```

Comment:
@qzhu-lbl @rgknox It appears this change is needed in master. Correct?

Reply:
yes